metrics/cgroups/metrics.go

```diff
@@ -56,54 +62,58 @@ func (c *Collector) Describe(ch chan<- *prometheus.Desc) {
 func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 	c.mu.RLock()
 	wg := &sync.WaitGroup{}
-	for id, cg := range c.cgroups {
+	for _, t := range c.cgroups {
 		wg.Add(1)
-		go c.collect(id, cg, ch, wg)
+		go c.collect(t.id, t.namespace, t.cgroup, ch, wg)
 	}
 	c.mu.RUnlock()
 	wg.Wait()
 }
 
-func (c *Collector) collect(id string, cg cgroups.Cgroup, ch chan<- prometheus.Metric, wg *sync.WaitGroup) {
+func (c *Collector) collect(id, namespace string, cg cgroups.Cgroup, ch chan<- prometheus.Metric, wg *sync.WaitGroup) {
 	defer wg.Done()
 	stats, err := cg.Stat(cgroups.IgnoreNotExist)
 	if err != nil {
 		logrus.WithError(err).Errorf("stat cgroup %s", id)
 		return
 	}
 	for _, m := range c.metrics {
-		m.collect(id, stats, c.ns, ch)
+		m.collect(id, namespace, stats, c.ns, ch)
 	}
 }
 
 // Add adds the provided cgroup and id so that metrics are collected and exported
-func (c *Collector) Add(id string, cg cgroups.Cgroup) error {
+func (c *Collector) Add(id, namespace string, cg cgroups.Cgroup) error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
-	if _, ok := c.cgroups[id]; ok {
+	if _, ok := c.cgroups[id+namespace]; ok {
```

Comment:
should we move this compound key generation into a function to make sure it is created consistently? Do we also need a separator between id and namespace to prevent edge cases that could cause collisions